fork/tests/run.sh

```
@@ -0,0 +1,236 @@
+
+printf '0\n%s\n' "$PIN_REV" >"$PUBLIC_CACHE_FILE"
+GIT_DIR="$WORKDIR/not-a-repo"
+assert_eq "cached_public_source_rev returns stale cache without blocking" "$(cached_public_source_rev)" "$PIN_REV"
+
+rm -f "$PUBLIC_CACHE_FILE"
+GIT_DIR="$WORKDIR/src-newer"
+GROK_FORK_SKIP_NETWORK=1
+assert_eq "cached_public_source_rev falls back to local SOURCE_REV" "$(cached_public_source_rev)" "$NEWER_REV"
+
+printf '%s\n1.0.5\n' "$(date +%s)" >"$PUBLIC_CACHE_FILE"
+GIT_DIR="$WORKDIR/src-same"
```

Shellcheck (0.11.0)

[warning] GIT_DIR appears unused. Verify use (or export if used externally).

(SC2034)